pynamodb/pagination.py

```diff
@@ -110,11 +110,7 @@ def next(self):
 
     @property
     def last_evaluated_key(self):
-        if self._first_iteration:
-            # Not started iterating yet: there cannot be a last_evaluated_key
-            return None
-
-        if self._index == self._count:
+        if self._first_iteration or self._index == self._count:
             # Entire page has been consumed: last_evaluated_key is whatever DynamoDB returned
```

Comment:
Can you update this comment to include why we fallback to the page iterator if `self._first_iteration` is true